### ISSUE TITLE
[Bug]: source failing Rushmoor Borough Council

### ISSUE THREAD

**Issue by slowgin** [mampfes/hacs_waste_collection_schedule#1485]
### I Have A Problem With:

Sensor configuration

### What's Your Problem

Rushmoor Borough Council looks to no longer be working , was fine for many months, 
the log shows HTTPError: 404 Client Error.  I cannot go to the url manual  looks not to have the page anymore
service provider website looks not to have the url. you can gat bid days from threre page search still

### Source (if relevant)

Rushmoor Borough Council

### Logs

```Shell
2023-12-05 20:50:08.825 ERROR (SyncWorker_3) [waste_collection_schedule.source_shell] fetch failed for source Rushmoor Borough Council:
Traceback (most recent call last):
File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch
entries = self._source.fetch()
^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/rushmoor_gov_uk.py", line 30, in fetch
r.raise_for_status()
File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://www.rushmoor.gov.uk/recycling-rubbish-and-environment/bins-and-recycling/download-or-print-your-bin-collection-calendar/?uprn=100060545016&weeks=16
```

### Relevant Configuration

```YAML
waste_collection_schedule:
    sources:
    - name: rushmoor_gov_uk
      args:
        uprn: "100060545016" 




sensor:
  - platform: waste_collection_schedule
    name: BinDay
    details_format: upcoming
    count: 15
    leadtime: 8
    value_template: '{{value.types|join(", ")}},on {{value.date.strftime("%a")}}'
    date_template:  'on {{value.date.strftime("%a")}}, {{value.date.strftime("%d.%m.%Y")}}'
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [ ] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [ ] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by panteLx** [comment]
This might be the new url (https://www.rushmoor.gov.uk/Umbraco/Api/BinLookUpWorkAround/Get?selectedAddress=10008776565&weeks=16) but i'm to dump to parse it :D tried my best.

With the following snipped you wont get the 404 but it needs to be parsed correctly :D
```
import requests
from waste_collection_schedule import Collection  # type: ignore[attr-defined]
from waste_collection_schedule.service.ICS import ICS

TITLE = "Rushmoor Borough Council"
DESCRIPTION = "Source for rushmoor.gov.uk services for Rushmoor, UK."
URL = "https://rushmoor.gov.uk"
TEST_CASES = {
    "GU14": {"uprn": "10008776565"},
}

ICON_MAP = {
    "Refuse": "mdi:trash-can",
    "Recycle": "mdi:recycle",
    "Garden": "mdi:leaf",
    "Food": "mdi:food-apple",
}

API_URL = "https://www.rushmoor.gov.uk/Umbraco/Api/BinLookUpWorkAround/Get?"


class Source:
    def __init__(self, uprn):
        self._uprn = uprn
        self._ics = ICS()

    def fetch(self):
        params = {"selectedAddress": self._uprn, "weeks": "16"}
        r = requests.get(API_URL, params=params)
        r.raise_for_status()

        dates = self._ics.convert(r.text)

        entries = []
        for d in dates:
            for wasteType in d[1].split("&"):
                wasteType = wasteType.replace("bin", "")
                wasteType = wasteType.strip()
                entries.append(
                    Collection(
                        d[0],
                        wasteType,
                        icon=ICON_MAP.get(wasteType),
                    )
                )
                # If wasteType is "Refuse" then add a second entry for "Garden"
                if wasteType == "Refuse":
                    entries.append(
                        Collection(
                            d[0],
                            "Garden",
                            icon=ICON_MAP.get("Garden"),
                        )
                    )

            # Always add Food as that is collected weekly
            entries.append(
                Collection(
                    d[0],
                    "Food",
                    icon=ICON_MAP.get("Food"),
                )
            )
        return entries

```

**Comment by 5ila5** [comment]
They removed the ICS files from their site I rewrote the source to now parse the json.
The implementation in #1503 will only return the next collection and no following ones. I do not know if the data chenges if collection dates are moved due to hollidays.

Fixed in #1503 
fix will be part of release 1.45.0 or is available now as master (git) version (HACS: 3-dot menu -> Redownload -> change version to master)
